src/teacher.py

```diff
@@ -77,23 +77,32 @@ def _answer_Demonstration_query(
 
 
 @dataclass
-class TeacherInteractionMonitor(utils.Monitor):
-    """Wraps the interaction between agent and teacher to include generating
-    and answering queries."""
+class TeacherInteractionMonitor(utils.VideoMonitor):
+    """A monitor that renders each state and action encountered and queries and
+    responses during interaction with the teacher.
+
+    The render_fn is generally env.render.
+    """
     _request: InteractionRequest
     _teacher: Teacher
+    _render_fn: Callable[[Optional[Action], Optional[str]], List[Image]]
```

Comment:
I would've expected that you don't need this or `_video` because the parent class already has those fields. Am I right?

Reply:
Also, isn't the argument order here different from how it's invoked?